File: lucafontanili.designpatterns.java/src/main/java/com/lucafontanili/designpatterns/structural/adapter/MediaPlayer.java
package com.lucafontanili.designpatterns.structural.adapter;

import com.lucafontanili.designpatterns.utilities.Constants;

public class MediaPlayer implements MediaPlayerInterface {

    private final AudioPlayerInterface audioPlayer = new AudioPlayer();

    @Override
    public void play(String fileName) {
	if (Constants.AUDIO_EXTENSIONS.contains(fileName.split("\\.")[1])) {
	    audioPlayer.playSong(fileName);
	} else {
	    LOGGER.info(new StringBuilder(32).append("Playing movie: ").append(fileName).toString());
	}

    }

}

File: lucafontanili.designpatterns.python/src/structural/adapter/MediaPlayer.py
'''
Created on Nov 5, 2016

@author: Admin
'''

from AudioPlayer import *

class MediaPlayerClass:
    
    _audio_player = AudioPlayerClass() 
    _audio_extensions = ['flac', 'mp3']
    
    def play(self, title):
        if title.split('.')[1] in self._audio_extensions:
            self._audio_player.play_song(title)
        else:
            print('Playing movie: ' + title)